pkg/router/functionHandler.go

```diff
@@ -221,18 +221,19 @@ func (roundTripper *RetryingRoundTripper) RoundTrip(req *http.Request) (*http.Re
 			req.URL.Scheme = roundTripper.serviceURL.Scheme
 			req.URL.Host = roundTripper.serviceURL.Host
 
-			functionNamespacedURL := "/" + fnMeta.Namespace + "/" + fnMeta.Name
-			defaultNamespacedURL := "/fission-function/" + fnMeta.Name
-			if strings.HasPrefix(req.URL.Path, functionNamespacedURL) || strings.HasPrefix(req.URL.Path, defaultNamespacedURL) {
-				req.URL.Path = "/"
-			} else {
-				if roundTripper.funcHandler.httpTrigger.Spec.Prefix != nil {
+			if roundTripper.funcHandler.httpTrigger != nil && roundTripper.funcHandler.httpTrigger.Spec.Prefix != nil {
+				functionNamespacedURL := "/" + fnMeta.Namespace + "/" + fnMeta.Name
+				defaultNamespacedURL := "/fission-function/" + fnMeta.Name
+				if strings.HasPrefix(req.URL.Path, functionNamespacedURL) || strings.HasPrefix(req.URL.Path, defaultNamespacedURL) {
+					req.URL.Path = "/"
```

Comment:
We need this block outside the if block as all functions might not have http triggers.